File: 6_sorting/max_product_of_three.py
def mul_triple(A, a, b, c):
    return A[a] * A[b] * A[c]

def solution(A):
    l = len(A)

    A.sort()

    candidate = mul_triple(A, -1, -2, -3)

    if A[0] * A[1] > 0:
        return max(candidate,
                   mul_triple(A, 0, 1, -1))
    return candidate

File: 06_sorting/max_product_of_three.cpp
#include "vector"
#include "algorithm"

using namespace std;

int solution(vector<int> &a) {
  int n = a.size();
  int ret;
  sort(a.begin(), a.end());
  ret = a[n-1] * a[n-2] * a[n-3];
  if (a[1] < 0) {
    return max(ret, a[0] * a[1] * a[n-1]);
  }
  return ret;
}
